### ISSUE TITLE
Bug: React Lazy Suspense with Errorboundary fails on older devices.

### ISSUE THREAD

**Issue by apoorvmote** [facebook/react#18383]
It is standard React Lazy wrapped with suspense that is wrapped with error boundary. 

I tested it on Windows 10 Chrome. It works.
I tested it on iPad Chrome it works. 
I tested it on Samsung Galaxy m 30s Chrome it works.

I tested on iPhone SE Chrome it **fails**.
I tested on iPhone 5s Chrome it **fails**. 
Also my customers have old android phones that is their primary device and it also **fails** on their device. 

There is no way to find error for **ME** on error boundary. There are no dev tools available on mobile devices and mobile evolution on Windows 10 Chrome always succeeds. But on **actual iPhone 5s & SE device it fails**. I know there are error logging apps but I don't know how to use one.

React version: 16.13.1

**Comment by bvaughn** [comment]
It doesn't look like this bug report has enough info for one of us to reproduce it.

Please provide a CodeSandbox (https://codesandbox.io/s/new), a link to a repository on GitHub, or provide a minimal code example that reproduces the problem. Screenshots or videos can also be helpful if they help provide context on how to repro the bug.

Here are some tips for providing a minimal example: https://stackoverflow.com/help/mcve

**Comment by apoorvmote** [comment]
I have created [public github repo](https://github.com/apoorvmote/lazy-suspense) for recreating issue. 

**It failed on production build served over https on actual iPhone 5s and iPhone SE device** not Desktop Chrome emulation of iPhone 5s/SE

**Comment by bvaughn** [comment]
What are the steps to reproduce the bug? What does it mean to "fail"?

**Comment by apoorvmote** [comment]
> What are the steps to reproduce the bug? 

- `npm run build`
- go to either PageA or PageB

I tested the app on my computer on dev environment it worked. So I published production version and started doing final testing on actual physical mobile devices. And it failed on iPhone 5s & iPhone SE. Worked on iPad & Samsung m 30s.

> What does it mean to "fail"?

When you go to any of the pages that are lazy loaded. You will see this message on test app. 

`Failed to load page because internet connection lost. Connect to the internet and try again. `

Basically there is error in `ErrorBoundary`. If you remove the `ErrorBoundary` that is wrapping `Suspense` then you get only white screen forever.

**Comment by gaearon** [comment]
>Failed to load page because internet connection lost. Connect to the internet and try again.

This kind of error doesn't sound like it has anything to do with React.

**Comment by apoorvmote** [comment]
@gaearon Please look at my code this is not error message. I wrote this custom string to deceive users into thinking its their fault the page is not loading. But in fact react has crashed. 

I was answering question how will you know that app has crashed. My custom fake error message will show up.

**Comment by apoorvmote** [comment]
@gaearon Please look at my code below
```
render() {

        if (this.state.hasError) {
            
            return (
                <main>
                    <h1>
                    Failed to load page because internet connection lost. Connect to the internet and try again. 
                    </h1>
                </main>
            )
        }

        return this.props.children
    }
```
It's fake error message.

**Comment by DrewVartanian** [comment]
@apoorvmote Have you figured out anything over the past couple of months?  I'm seeing the same thing when using an iPad 4.

**Comment by apoorvmote** [comment]
Well I just told my client that its limitation from react. So forced them to use newer mobile devices. I don't think react people want to support older devices. These bugs will never get fixed.
